azext_iot/iothub/providers/state.py

```diff
@@ -276,22 +452,70 @@ def upload_module_identity(self, identity: dict):
 
+    def delete_all_certificates(self):
+        certificates = self.cli.invoke("iot hub certificate list --hub-name {} -g {}".format(self.hub_name, self.rg)).as_json()
+        for i in tqdm(range(len(certificates["value"])), desc="Deleting certificates from destination hub", file=sys.stdout):
+            c = certificates["value"][i]
+            self.cli.invoke("iot hub certificate delete --name {} --etag {} --hub-name {} -g {}".format(c["name"], c["etag"],
+                                                                                                        self.hub_name, self.rg))
+
+    def delete_all_endpoints(self):
+        endpoints = self.cli.invoke(f"iot hub routing-endpoint list --hub-name {self.hub_name} -g {self.rg}").as_json()
+        eventHubs = endpoints["eventHubs"]
+        serviceBusQueues = endpoints["serviceBusQueues"]
+        serviceBusTopics = endpoints["serviceBusTopics"]
+        storageContainers = endpoints["storageContainers"]
+
+        num_endpoints = len(eventHubs) + len(serviceBusQueues) + len(serviceBusTopics) + len(storageContainers)
+        pbar = tqdm(total=num_endpoints, desc="Deleting endpoints from destination hub", file=sys.stdout)
+
+        for ep in eventHubs:
+            self.cli.invoke("iot hub routing-endpoint delete --hub-name {} -g {} --endpoint-name {} --endpoint-type \
+                eventhub".format(self.hub_name, self.rg, ep["name"]))
+            pbar.update(1)
+        for ep in serviceBusQueues:
+            self.cli.invoke("iot hub routing-endpoint delete --hub-name {} -g {} --endpoint-name {} --endpoint-type \
+                servicebusqueue".format(self.hub_name, self.rg, ep["name"]))
+            pbar.update(1)
+        for ep in serviceBusTopics:
+            self.cli.invoke("iot hub routing-endpoint delete --hub-name {} -g {} --endpoint-name {} --endpoint-type \
+                servicebustopic".format(self.hub_name, self.rg, ep["name"]))
+            pbar.update(1)
+        for ep in storageContainers:
+            self.cli.invoke("iot hub routing-endpoint delete --hub-name {} -g {} --endpoint-name {} --endpoint-type \
+                azurestoragecontainer".format(self.hub_name, self.rg, ep["name"]))
+            pbar.update(1)
+
+        pbar.close()
+
+    def delete_all_routes(self):
+        routes = self.cli.invoke(f"iot hub route list --hub-name {self.hub_name} -g {self.rg}").as_json()
+        for i in tqdm(range(len(routes)), desc="Deleting routes from destination hub", file=sys.stdout):
+            route = routes[i]
+            self.cli.invoke(f"iot hub route delete --hub-name {self.hub_name} -g {self.rg} --name {route['name']}")
+
     def upload_state(self, filename: str, replace: Optional[bool] = None):
         '''
         Uses device info from file to recreate the devices
         '''
 
         if replace:
+            if self.include_control_plane:
+                with self.suppress_stderr():
+                    self.delete_all_routes()
```

Comment:
If stderr is suppressed rather than being captured, if something fails in these operations within the context scope, will the user be notified on what failed (like API messages which may give hints for resolution)?